src/map/battle.cpp

```diff
@@ -7224,8 +7224,8 @@ struct Damage battle_calc_magic_attack(struct block_list *src,struct block_list
 					case NPC_CLOUD_KILL:
 						skillratio += -100 + 50 * skill_lv;
 						break;
-					case SO_VARETYR_SPEAR: //MATK [{( Endow Tornado skill level x 50 ) + ( Caster INT x Varetyr Spear Skill level )} x Caster Base Level / 100 ] %
-						skillratio += -100 + sstatus->int_ / 6 * skill_lv + ((sd) ? pc_checkskill(sd, SA_LIGHTNINGLOADER) * 50 : 0); // !TODO: Confirm new formula
+					case SO_VARETYR_SPEAR:
+						skillratio += -100 + (2 * sstatus->int_ + 150 * ((sd ? pc_checkskill(sd, SO_STRIKING) : 0) + (sd ? pc_checkskill(sd, SA_LIGHTNINGLOADER) : 0)) + sstatus->int_ * skill_lv / 2) / 3;
```

Comment:
The short if for sd could be removed. The first check inside pc_checkskill is if sd is nullptr and return 0 in this case.